docs/articles/documentation/test-api/test-code-structure.md

```diff
@@ -572,3 +573,163 @@ test('Fixture2Test3', () => {});
+`path`    | String | The path to the JavaScript file whose content should be injected. Relative paths resolve from the directory that contains the current test file.
+
+**Example**
+
+```js
+fixture.clientScripts('assets/jquery.js');
+fixture.clientScripts({ path: 'assets/jquery.js' });
+```
+
+> You cannot combine the `path`, [module](#specify-the-module-name) and [content](#specify-the-script-code) properties.
+
+### Specify the Module Name
+
+Specify the module name to inject the module's content into the tested pages. Use a string or an object with the `module` property.
+
+```text
+fixture.clientScripts( module | { module } )
+```
+
+```text
+test.clientScripts( module | { module } )
+```
+
+Argument  | Type   | Description
+--------- | ------ | ---------------------------------------------------------------------------
+`module`  | String | The name of the module whose content should be injected. TestCafe uses [require.resolve](https://nodejs.org/api/modules.html#modules_require_resolve_request_options) to resolve the module location with Node.js machinery.
+
+**Example**
+
+```js
+fixture.clientScripts('lodash');
+fixture.clientScripts({ module: 'async' });
+```
+
+> You cannot combine the `module`, [path](#specify-the-path-to-a-javascript-file) and [content](#specify-the-script-code) properties.
+
+### Specify the Script Code
+
+You can provide the injected script as a string with JavaScript code. Pass an object with the `content` property to do this.
+
+```text
+fixture.clientScripts( { content } )
+```
+
+```text
+test.clientScripts( { content } )
+```
+
+Argument  | Type   | Description
+--------- | ------ | ----------------------------------------
+`content` | String | JavaScript code that should be injected.
+
+**Example**
+
+```js
+const mockDate = `
+    Date.prototype.getTime = function () {
+        return 42;
+    };
+`;
+
+test.clientScripts({ content: mockDate });
+```
+
+> You cannot combine the `content`, [path](#specify-the-path-to-a-javascript-file) and [module](#specify-the-module-name) properties.
+
+### Provide Scripts for Particular Pages
+
+You can also specify pages into which a script should be injected. This is helpful when you need to use custom mocks on particular pages and preserve the default behavior everywhere else.
+
+To specify target pages for a script, add the `page` property to the object you pass to `clientScripts`.
+
+```text
+fixture.clientScripts({ page, path | module | content })
+```
+
+```text
+test.clientScripts({ page, path | module | content })
+```
+
+Property  | Type                | Description
+--------- | ------------------- | ---------------------------------------------------------------------------
+`page`    | String &#124; RegExp | Specify a page URL to add scripts to a single page, or a regular expression to add scripts to pages whose URLs match this expression. An empty string adds scripts to all pages visited during the test/fixture.
```

Comment:
> An empty string adds scripts to all pages visited during the test/fixture.

It's not true.

If 'page' property is not specified then script will be added to all pages visited during test/fixture.



``page`    | String &#124; RegExp` - possible all variants from https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/select-requests-to-be-handled-by-the-hook.html.
May we rename/move this topic and use here as link?

Reply:
OK for the empty value.

Regarding possible options: I understand the `String` or `RegExp`, but the rest of them - do they make sense for script injection? An object that includes `method` or `isAjax` properties, a predicate that receives `headers` and `body` - it's so request-specific...

Reply:
It's common concept - request filtering.
If request satisfies the rule then we will do the certain action: log request, mock request or inject script.

Reply:
I don't get what you are talking about.
You said that the entire request filtering API applies to injecting scripts as well. I wonder how, provided that it exposes thing like `isAjax`, `headers`, `body`, etc.